huaweicloud/services/acceptance/apig/data_source_huaweicloud_apig_api_basic_configurations_test.go

```diff
@@ -0,0 +1,436 @@
+					resource.TestCheckOutput("is_env_id_filter_useful", "true"),
+					resource.TestCheckOutput("is_env_name_filter_useful", "true"),
+					resource.TestCheckOutput("is_backend_type_filter_useful", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceApiBasicConfigurations_base() string {
+	name := acceptance.RandomAccResourceName()
+
+	return fmt.Sprintf(`
+%[1]s
+
+resource "huaweicloud_apig_custom_authorizer" "frontEnd" {
+  instance_id      = huaweicloud_apig_instance.test.id
+  name             = "%[2]s_front"
+  function_urn     = huaweicloud_fgs_function.test.urn
+  function_version = "latest"
+  type             = "FRONTEND"
+}
+
+resource "huaweicloud_apig_api" "test" {
+  instance_id             = huaweicloud_apig_instance.test.id
+  group_id                = huaweicloud_apig_group.test.id
+  name                    = "%[2]s"
+  type                    = "Public"
+  request_protocol        = "HTTP"
+  request_method          = "GET"
+  request_path            = "/{%[2]s}"
+  security_authentication = "AUTHORIZER"
+  matching                = "Exact"
+  authorizer_id           = huaweicloud_apig_custom_authorizer.frontEnd.id
+  cors                    = true
+  description             = "Updated by script"
+  
+  request_params {
+    name     = "%[2]s"
+    type     = "NUMBER"
+    location = "PATH"
+    required = true
+    maximum  = 200
+    minimum  = 0
+  }
+
+  web {
+    path             = "/"
+    vpc_channel_id   = huaweicloud_apig_vpc_channel.test.id
+    request_method   = "GET"
+    request_protocol = "HTTP"
+    timeout          = 30000
+    retry_count      = 1
+  }
+}
+
+resource "huaweicloud_apig_environment" "test" {
+  instance_id = huaweicloud_apig_instance.test.id
+  name        = "%[2]s"
+}
+
+resource "huaweicloud_apig_api_publishment" "test" {
+  instance_id = huaweicloud_apig_instance.test.id
+  env_id      = huaweicloud_apig_environment.test.id
+  api_id      = huaweicloud_apig_api.test.id
+}
+
+`, testAccApi_base(name), name)
+}
+
+func testAccDataSourceApiBasicConfigurations_basic() string {
+	return fmt.Sprintf(`
+%[1]s
+
+data "huaweicloud_apig_api_basic_configurations" "test" {
+  depends_on = [
+    huaweicloud_apig_api_publishment.test
+  ]
+
+  instance_id = huaweicloud_apig_instance.test.id
+}
+
+# Filter by ID
+locals {
+  api_id = data.huaweicloud_apig_api_basic_configurations.test.configurations[0].id
+}
+
+data "huaweicloud_apig_api_basic_configurations" "filter_by_id" {
+  depends_on = [
+    huaweicloud_apig_api_publishment.test
+  ]
+
+  instance_id = huaweicloud_apig_instance.test.id
+  api_id      = local.api_id
+}
+
+locals {
+  id_filter_result = [
+    for v in data.huaweicloud_apig_api_basic_configurations.filter_by_id.configurations[*].id : v == local.api_id
+  ]
+}
+
+output "is_id_filter_useful" {
+  value = length(local.id_filter_result) > 0 && alltrue(local.id_filter_result)
+}
+
+# Filter by name (fuzzy search)
```

Comment:
please test whether the corresponding parameter (name) supports exact match.

Reply:
fixed